client/shipping-label/views/steps/rates/list.js

```diff
@@ -1,61 +1,80 @@
 import React, { PropTypes } from 'react';
 import Dropdown from 'components/dropdown';
+import Notice from 'components/notice';
 import { translate as __ } from 'lib/mixins/i18n';
+import { sprintf } from 'sprintf-js';
+import get from 'lodash/get';
+
+const renderRateNotice = ( show ) => {
+	if ( show ) {
+		return (
+			<Notice
+				className="rate-not-available"
+				icon="info-outline"
+				showDismiss={ false }
+				text={ __( 'The service and rate chosen by the customer at checkout is not available. Please choose another.' ) }
+			/>
+		);
+	}
+};
 
 const ShippingRates = ( {
 		id,
 		selectedRates,
 		availableRates,
 		packages,
 		updateRate,
-		dimensionUnit,
-		weightUnit,
 		currencySymbol,
 		errors,
+		showRateNotice,
 	} ) => {
-	const renderTitle = ( pckg ) => {
-		return `${pckg.length} ${dimensionUnit} x ${pckg.width} ${dimensionUnit} x ${pckg.height} ${dimensionUnit} (${pckg.weight} ${weightUnit})`;
+	const renderTitle = ( pckg, idx ) => {
+		if ( 1 === packages.length ) {
+			return __( 'Choose rate' );
+		}
+		return sprintf( __( 'Choose rate: Package %(index)d' ), { index: idx + 1 } );
 	};
 
 	const renderSinglePackage = ( pckg, index ) => {
-		const selectedRate = selectedRates[ index ] || '';
-
+		const selectedRate = selectedRates[ pckg.id ] || '';
+		const packageRates = get( availableRates, [ pckg.id, 'rates' ], [] );
 		const valuesMap = { '': __( 'Select one...' ) };
-		Object.keys( availableRates[ index ] ).forEach( ( serviceId ) => {
-			const rateObject = availableRates[ index ][ serviceId ];
-			valuesMap[ serviceId ] = rateObject.name + ' (' + currencySymbol + rateObject.rate.toFixed( 2 ) + ')';
+
+		packageRates.forEach( ( rateObject ) => {
+			valuesMap[ rateObject.service_id ] = rateObject.title + ' (' + currencySymbol + rateObject.rate.toFixed( 2 ) + ')';
 		} );
 
 		return (
-			<li key={ index }>
+			<div key={ index }>
 				<Dropdown
 					id={ id + '_' + index }
 					valuesMap={ valuesMap }
-					title={ renderTitle( pckg ) }
+					title={ renderTitle( pckg, index ) }
 					value={ selectedRate }
-					updateValue={ ( value ) => updateRate( index, value ) }
-					error={ errors[ index ] } />
-			</li>
+					updateValue={ ( value ) => updateRate( pckg.id, value ) }
+					error={ errors[ pckg.id ] } />
+			</div>
 		);
 	};
 
 	return (
-		<ul>
+		<div>
+			{ renderRateNotice( showRateNotice ) }
 			{ packages.map( renderSinglePackage ) }
-		</ul>
+		</div>
 	);
 };
 
 ShippingRates.propTypes = {
 	id: PropTypes.string.isRequired,
-	selectedRates: PropTypes.array.isRequired,
-	availableRates: PropTypes.array.isRequired,
+	selectedRates: PropTypes.object.isRequired,
```

Comment:
This is whatever the merchant has selected in the dialog, right?  And not necessarily what the customer selected.  Let's add a comment maybe clarifying that for developers new to the code.